woocommerce/woocommerce-gutenberg-products-block
# Hook up shipping methods to API

This PR hooks up the shipping methods list in the _Cart_ and _Checkout_ components with the API endpoints introduced in #1344.

### Screenshots

![Peek 2020-01-19 12-32](https://user-images.githubusercontent.com/3616980/72680232-cebadb80-3ab7-11ea-96d4-a6d297a656b7.gif)

### How to test the changes in this Pull Request:

1. Make sure you have different shipping methods set up and create a post with a _Checkout_ block.
2. Fill the checkout _Shipping_ step and verify shipping rates update accordingly.
3. Test again setting some package rules using the [WooCommerce Advanced Shipping Packages](https://woocommerce.com/products/woocommerce-advanced-shipping-packages/#) extension.

### Follow-ups
I left some `@todo` comments in the code with tasks to be fixed in the future and I created issues for them:
* Radio inputs are a bit misaligned, but I think it's better to fix that when we update its styling to match `@wordpress/components` (#1556).
* Display product name and quantity before package options (#1554).
* Add UI for loading state of shipping rates (#1555).
* Display currently selected location in Cart no results message (#1606).

### To do in this PR
* [x] Rebase & fix conflicts.
* [x] Add debounce (d22af0be78ea4ce64e534c524fdc5301409ddc6c).
* [x] Make endpoint return an error if country is invalid (602141d900caecd6234d3da80b9bbfa2aca1f601).
* [x] Don't display radio input when there is only one option (c6130f09033bbfb4d8797f43cc2351616196aca9).
